arm-machinelearning/2016-05-01-preview/swagger/commitmentPlans.json

```diff
@@ -0,0 +1,720 @@
+        "parameters": [
+          {
+           "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "name": "resourceGroupName",
+            "description": "The resource group name.",
+            "in": "path",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "planName",
+            "description": "The commitment plan name.",
+            "in": "path",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "$skipToken",
+            "description": "Continuation token for pagination.",
+            "in": "query",
+            "required": false,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/APIVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/PaginatedResult[PlanUsageHistory]"
+            }
+          }
+        }
+      }
+    }
+  },
+  "parameters": {
+    "SubscriptionIdParameter": {
+      "name": "subscriptionId",
+      "description": "Azure Subscription ID.",
+      "in": "path",
+      "type": "string",
+      "required": true
+    },
+    "APIVersionParameter": {
+      "in": "query",
+      "name": "api-version",
+      "type": "string",
+      "description": "The versiong of the Microsoft.MachineLearning resource provider API to be used.",
+      "required": true
+    }
+  },
+  "definitions": {
+    "Resource": {
+      "x-ms-azure-resource": true,
+      "description": "Common properties of an ARM resource.",
+      "required": [
+        "location"
+      ],
+      "properties": {
+        "id": {
+          "description": "Resource Id",
+          "type": "string",
+          "readOnly": true
+        },
+        "name": {
+          "description": "Resource Name",
+          "type": "string"
+        },
+        "location": {
+          "description": "Resource Location",
+          "type": "string"
+        },
+        "type": {
+          "description": "Resource type",
+          "type": "string",
+          "readOnly": true
+        },
+        "tags": {
+          "description": "Resource tags",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "CommitmentAssociation": {
+      "type": "object",
+      "description": "Represents the association between a commitment plan and some other resource, such as a Machine Learning web service.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "properties": {
+        "etag": {
+          "type": "string",
+          "description": "An entity tag used to enforce optimistic concurrency."
+        },
+        "properties": {
+          "$ref": "#/definitions/CommitmentAssociationProperties"
```

Comment:
Dont you want to flatten them. This makes life easy for the end user. They dont have to access CommitmentAssociation.properties.foo. They can access it like this `CommitmentAssociation.foo` if you add the extension `"x-ms-client-flatten":true` over here. The generated code will make sure to [de]serialize it correctly.